### ISSUE TITLE
Apply for Open Source programs from CDN providers

### ISSUE THREAD

**Issue by ameukam** [kubernetes/steering#239]
### Problem Statement
The Kubernetes project produces and distributes various artifacts used by the community. Currently system packages and binaries are hosted on Google's infrastructure. They should be hosted within community infrastructure that primarily rely on a cloud computing provider. Once the migration of those artifacts is done, we may face some issues (we faced the same issues with the migration k8s.gcr.io) :

- high cost of being the only responsible for distribution. 
- high cost in terms of maintenance for underlying infrastructure. 
- concerns of latency or access in downloading for those 

### Proposed Solution
I would like to propose we use a CDN provider for distribution. The proposal is not new; it's already done by many Open source projects for a long time now:
- Linux Kernel  Project: https://www.kernel.org/introducing-fastly-cdn.html
- The Apache Foundation: https://apache.org/history/mirror-history.html
- NixOS project: https://nixos.wiki/wiki/Maintainers:Fastly

### Cost
The cost is supposed to be supported by the plan we apply for but we still need to look at the details of those different plans.

### Open Questions
- [ ] 

### Next Steps
- [ ] Apply to one of those programs
  - https://blog.cloudflare.com/cloudflare-open-source-your-upgrade-is-on-the-house/
  - https://www.fastly.com/open-source/

EDIT: I don't think I can apply to for example The Fastly program on the behalf of the community  so I would like one of SC members to apply to it.

**Comment by ameukam** [comment]
/assign @dims
AS SIG k8s-infra chair

**Comment by dims** [comment]
we talked about this in the sig-k8s-infra meeting as well. essentially Arnaud will update the request here to reflect that one of us (probably me) will fill the form with say fastly and explore what they offer, how this works and what they want in return as well.

**Comment by ameukam** [comment]
I still don't know exactly what is offered in terms of features in the various Open Source programs. The request is made in order to engage in conversation with at least one service provider (possibly Fastly due to the partnership with the LF) to identify and clarify the technical requirements/constraints before engaging.

**Comment by justaugustus** [comment]
👀 👀 👀 @kubernetes/release-engineering

**Comment by ameukam** [comment]
Still in progress.

/remove-lifecycle stale

**Comment by dims** [comment]
@kubernetes/steering-committee Arnaud (@ameukam) has been able to get good results from the proof-of-concept in https://github.com/kubernetes/k8s.io/issues/3596 . So i will open up a service desk ticket that will help us rope in @caniszczyk and others from CNCF to talk to CloudFlare

**Comment by dims** [comment]
This is back in @kubernetes/steering-committee court @ameukam 

```
Amye Scavarda Perrin
Today 1:45 PM
So our preference for things like this is to have the projects reach out directly to the provider, and involve us if there needs to be some sort of zero-balance account billing.

Here, it looks like Cloudflare just needs an email to [os-sponsorship@cloudflare.com](mailto:os-sponsorship@cloudflare.com) with a link to the project’s landing page, repo, and a description of what engineering tools or resources your project provides to the developer community. 

Can the steering committee run with that? It makes it a lot easier if you all run into issues down the road because they have a primary contact within the project.

If not, we can work out a way to connect with them, it’s just easier to have the project start the process.

```

**Comment by ameukam** [comment]
> This is back in @kubernetes/steering-committee court @ameukam
> 
> ```
> Amye Scavarda Perrin
> Today 1:45 PM
> So our preference for things like this is to have the projects reach out directly to the provider, and involve us if there needs to be some sort of zero-balance account billing.
> 
> Here, it looks like Cloudflare just needs an email to [os-sponsorship@cloudflare.com](mailto:os-sponsorship@cloudflare.com) with a link to the project’s landing page, repo, and a description of what engineering tools or resources your project provides to the developer community. 
> 
> Can the steering committee run with that? It makes it a lot easier if you all run into issues down the road because they have a primary contact within the project.
> 
> If not, we can work out a way to connect with them, it’s just easier to have the project start the process.
> ```

With the probability that we may be listed in https://developers.cloudflare.com/sponsorships/, can you check with CNCF if there is any concern about Trademark logo ?

**Comment by dims** [comment]
@ameukam no concerns about us getting listed there on that url.

**Comment by vielmetti** [comment]
Showing up here re the Equinix Metal open source program, as referenced in the discussion from the 2022-11-09 sig-k8s-infra meeting.

**Comment by BenTheElder** [comment]
We are in fastly's CDN program now, announcements pending.

**Comment by BenTheElder** [comment]
https://kubernetes.io/blog/2023/06/09/dl-adopt-cdn/
